File: test/fixtures/jparam/ios/Sources/JparamPlugin.swift
import Foundation
import Capacitor

@objc(JparamPlugin)
public class JparamPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "JparamPlugin"
    public let jsName = "Jparam"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "list", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "count", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "describe", returnType: CAPPluginReturnPromise)
    ]

    private var items: [String] = ["a", "b", "c"]

    @objc func list(_ call: CAPPluginCall) {
        call.resolve([
            "items": items
        ])
    }

    @objc func count(_ call: CAPPluginCall) {
        call.resolve([
            "count": items.count
        ])
    }

    @objc func describe(_ call: CAPPluginCall) {
        guard let label = call.getString("label") else {
            call.reject("label required")
            return
        }
        call.resolve([
            "label": label,
            "active": items.contains(label)
        ])
    }
}

File: test/fixtures/jparam/android/src/main/java/com/example/jparam/JparamPlugin.java
package com.example.jparam;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Jparam")
public class JparamPlugin extends Plugin {

    private java.util.List<String> items = new java.util.ArrayList<>(java.util.Arrays.asList("a", "b", "c"));

    @PluginMethod
    public void list(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("items", new java.util.ArrayList<>(items));
        call.resolve(ret);
    }

    @PluginMethod
    public void count(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("count", items.size());
        call.resolve(ret);
    }

    @PluginMethod
    public void describe(PluginCall call) {
        String label = call.getString("label");
        if (label == null) {
            call.reject("label required");
            return;
        }
        JSObject ret = new JSObject();
        ret.put("label", label);
        ret.put("active", items.contains(label));
        call.resolve(ret);
    }
}
